src/testing/tests/precompiles/ecmul.rs

```diff
@@ -0,0 +1,149 @@
+use super::*;
+use zk_evm_abstractions::auxiliary::*;
+use zk_evm_abstractions::queries::*;
+use zk_evm_abstractions::vm::*;
+use zkevm_opcode_defs::system_params::*;
+use zkevm_opcode_defs::PrecompileCallABI;
+
+fn fill_memory<M: Memory>(
+    x1: [u8; 32],
+    y1: [u8; 32],
+    s: [u8; 32],
+    page: u32,
+    memory: &mut M,
+) -> u16 {
+    let mut location = MemoryLocation {
+        page: MemoryPage(page),
+        index: MemoryIndex(0),
+        memory_type: MemoryType::Heap,
+    };
+    let query = MemoryQuery {
+        timestamp: Timestamp(0u32),
+        location,
+        value: U256::from_big_endian(&x1),
+        rw_flag: true,
+        value_is_pointer: false,
+    };
+    let _ = memory.execute_partial_query(0, query);
+
+    location.index.0 += 1;
+    let query = MemoryQuery {
+        timestamp: Timestamp(0u32),
+        location,
+        value: U256::from_big_endian(&y1),
+        rw_flag: true,
+        value_is_pointer: false,
+    };
+    let _ = memory.execute_partial_query(1, query);
+
+    location.index.0 += 1;
+    let query = MemoryQuery {
+        timestamp: Timestamp(0u32),
+        location,
+        value: U256::from_big_endian(&s),
+        rw_flag: true,
+        value_is_pointer: false,
+    };
+    let _ = memory.execute_partial_query(2, query);
+
+    3 as u16
```

Comment:
nit, but don't need to explicitly cast here as it's implied from the return type